seatunnel-formats/seatunnel-format-text/src/main/java/org/apache/seatunnel/format/text/splitor/CsvLineSplitor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.format.text.splitor;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CsvLineSplitor implements TextLineSplitor, Serializable {
+    private CSVFormat format;
+
+    @Override
+    public String[] spliteLine(String line, String splitor) {
+        if (Objects.isNull(format)) {
```

Comment:
If parsing nested types, the delimiter will change, and when entering the next layer of recursion, the delimiter will still be the previous layer's delimiter. So I think there should create a map to store the formats of delimiters. If format == null and create.

Reply:
I agree, but usually the CSV format doesn't support nested types I think.  Can we simply extract that field and let user to do the further data processing(like deserializing the JSON or XML)?

Reply:
In most production cases, if you want to use CSV to save nested types, you will follow the invisible character principle created by hive. The first layer of CSV is a comma, the second layer may be \u0002, and the third layer may be \u0003. In fact, when parsing each layer, only the delimiter needs to be specified because there will be no duplicate delimiters when parsing each layer.